src/data/types/range.ts

```diff
@@ -1,19 +1,33 @@
 import { Tagged } from "../../cbor";
 import { SurrealDbError } from "../../errors";
+import { equals } from "../../util/equals";
 import { toSurrealqlString } from "../../util/to-surrealql-string";
 import { TAG_BOUND_EXCLUDED, TAG_BOUND_INCLUDED } from "../cbor";
+import { Value } from "../value";
 import {
 	type RecordIdValue,
 	escape_id_part,
 	escape_ident,
 	isValidIdPart,
 } from "./recordid";
 
-export class Range<Beg, End> {
+export class Range<Beg, End> extends Value {
 	constructor(
 		readonly beg: Bound<Beg>,
 		readonly end: Bound<End>,
-	) {}
+	) {
+		super();
+	}
+
+	equals(other: unknown): boolean {
+		if (!(other instanceof Range)) return false;
+		if (this.beg?.constructor !== other.beg?.constructor) return false;
+		if (this.end?.constructor !== other.end?.constructor) return false;
+		return (
+			equals(this.beg?.value, other.beg?.value) &&
+			equals(this.end?.value, other.end?.value)
```

Comment:
Don't the bound values also need to implement `Value` and `.equals()` for this to work?

Reply:
It delegates back to the global equals function which also compares primitives, arrays, and objects, so no!